src/main/scala/com/daml/projection/javadsl/BatchSource.scala

```diff
@@ -33,18 +38,29 @@ object BatchSource {
   /**
    * Creates a [[BatchSource]] from existing batches, useful for testing purposes.
    */
-  def create[E](batches: java.lang.Iterable[Batch[E]]): BatchSource[E] = S.BatchSource(batches.asScala.toList).toJava
+  def create[E](
+      batches: java.lang.Iterable[Batch[E]],
+      templateIdFromEvent: java.util.function.Function[E, ju.Optional[JIdentifier]],
+      partySetFromEvent: java.util.function.Function[E, ju.Set[String]]): BatchSource[E] =
```

Comment:
```suggestion
      stakeholdersFromEvent: java.util.function.Function[E, ju.Set[String]]): BatchSource[E] =
```

Reply:
> The stakeholders are the union of the signatories and the observers of the contract.

Reply:
Change the names of these in other instances as well.

Reply:
Actually stakeholders would not be correct for TreeEvent and ExercisedEvent, rather call it parties

Reply:
will change it to partiesFromEvent